### PR TITLE
Scheduling feature

### PR DESCRIPTION
**Pull request by enewton3**

### CODE REVIEW

File: src/utils/scheduling.js [hunk 1]
```diff
@@ -33,10 +48,215 @@ const ALL_MEME_CAPTIONS = [
...
+      .addComponents(
+        new ButtonBuilder()
+          .setCustomId(`8p#${day.id}`)
+          .setLabel("8:00pm")
+          .setStyle(ButtonStyle.Primary)
+      );
+
+    const embed = new EmbedBuilder().addFields(
+      BLANK_EMBED_FIELD,
+      { name: `6:00pm`, value: "No gamers", inline: true },
```
Comment (sammyers) [inline src/utils/scheduling.js:152]: do you think we need more options? sometimes we start later than 8 or maybe occasionally earlier than 6

File: src/utils/scheduling.js [hunk 2]
```diff
@@ -33,10 +48,215 @@ const ALL_MEME_CAPTIONS = [
...
+
+  interaction.message.edit({ embeds: [embed] });
+  interaction.deferUpdate();
+};
+
+const eventCreatorJob = async (channelId, client) => {
+  const currentWeek = getCurrentWeek();
+  const days = await getUsersForGameNights(currentWeek);
+
+  const guild = await client.guilds.fetch(`${process.env.SERVER_ID}`);
```
Comment (sammyers) [inline src/utils/scheduling.js]: seems like we do this logic of fetching the guild and the channel a few times, can we put it in a utility function?

File: src/utils/scheduling.js [hunk 3]
```diff
@@ -33,10 +48,215 @@ const ALL_MEME_CAPTIONS = [
...
+  const channel = await guild.channels.fetch(`${channelId}`);
+
+  Object.entries(days).forEach((day) => {
+    if (!day) return;
+    const [dayId, dayObj] = day;
+    if (!dayObj.times) return;
+    const times = Object.entries(dayObj.times);
```
Comment (sammyers) [inline src/utils/scheduling.js:219]: yikes this whole sequence is near impossible to follow without TypeScript

File: src/utils/scheduling.js [hunk 4]
```diff
@@ -33,10 +48,215 @@ const ALL_MEME_CAPTIONS = [
...
+    times.forEach(async (timeObj) => {
+      const [time, users] = timeObj;
+      //change me when push to prod
+      if (Object.entries(users).length >= 3) {
+        const now = DateTime.now();
+
+        const weekdayNum = weekdaysMap[dayObj.day];
+        const daysTillWeekday = (weekdayNum - now.weekday + 7) % 7;
+
+        const timeMap = { "6:00pm": 18, "7:00pm": 19, "8:00pm": 20 };
```
Comment (sammyers) [inline src/utils/scheduling.js:229]: wait are we storing the times in the DB as literal strings "8:00pm" and so on? don't love that, I feel like it's adding some unnecessary parsing and formatting complexity

File: src/utils/scheduling.js [hunk 5]
```diff
@@ -33,10 +48,215 @@ const ALL_MEME_CAPTIONS = [
...
+      DateTime.DATETIME_FULL
+    )}`
+  );
+};
+
+const schedulerJob = async (channelId, client) => {
+  console.log("Running Scheduler Job");
+  const days = await getUsersForGameNights(getCurrentWeek());
+  const daysArray = Object.entries(days)
+    .filter((v) => (v[1].users ? true : false))
```
Comment (sammyers) [inline src/utils/scheduling.js:106]: a lot of these lines with the tuple indexing are kind of hard to follow, can we utilize some named destructuring to make the data types clearer?

File: src/utils/scheduling.js [hunk 6]
```diff
@@ -33,10 +48,215 @@ const ALL_MEME_CAPTIONS = [
...
+
+  console.log(
+    `scheduler job scheduled for ${schedulerJobDate.toLocaleString(
+      DateTime.DATETIME_FULL
+    )}`
+  );
+  scheduler.start();
+
+  const daysTillFriday = (friday - today + 7) % 7;
+  // const fridayJobDate = now.plus({ seconds: 40 });
```
Comment (sammyers) [inline src/utils/scheduling.js]: do we need this line

File: src/utils/scheduling.js (hunk 4 above)
Comment (sammyers) [reply src/utils/scheduling.js:229]: this mapping also feels a bit brittle if we end up implementing more flexible options

File: src/main.js [hunk 7]
```diff
@@ -100,6 +107,12 @@ client.on("interactionCreate", async (interaction) => {
     if (interaction.commandName === "itstimetoduel") {
       await handleSchedulingInteraction(interaction, client);
       interaction.channel.createMessageComponentCollector;
```
Comment (sammyers) [inline src/main.js]: can you delete this line haha I don't know why it's still there

File: src/utils/scheduling.js [hunk 8]
```diff
@@ -33,10 +48,215 @@ const ALL_MEME_CAPTIONS = [
...
+};
+
 const getMemeCaptions = (meme) => {
   return sample(
     ALL_MEME_CAPTIONS.filter((captions) => captions.length === meme.box_count)
   );
 };
 
-module.exports = { getMemeCaptions, MAX_BOXES };
+const schedulerJobScheduler = (channelId, client) => {
```
Comment (sammyers) [inline src/utils/scheduling.js:68]: Could you add some comments here on how the timing/experience works for the scheduling flow? it's not super apparent to me just from reading the code when the scheduling prompt is sent out for each day and when the event creator runs. Does it ask for responses on thursday for all of the potential weekend days and then create the events on friday?

**Review by sammyers — COMMENTED** [review]
Overall structure of this looks good but I have some questions about things

File: src/utils/scheduling.js (hunk 2 above)
Comment (enewton3) [reply src/utils/scheduling.js]: We can do this, means passing the client around a lot more I think
